front/components/agent_builder/capabilities/AgentBuilderCapabilitiesBlock.tsx

```diff
@@ -62,8 +75,71 @@ function ActionCard({
   );
 }
 
+function MaxStepsPerRunSettings() {
+  const { owner } = useAgentBuilderContext();
+  const { hasFeature, isFeatureFlagsLoading } = useFeatureFlags({
+    workspaceId: owner.sId,
+  });
+  const { field } = useController<AgentBuilderFormData, "maxStepsPerRun">({
+    name: "maxStepsPerRun",
+  });
+
+  const hasExtendedFeature = hasFeature("extended_max_steps_per_run");
+  const maxLimit = hasExtendedFeature
+    ? EXTENDED_MAX_STEPS_USE_PER_RUN_LIMIT
+    : MAX_STEPS_USE_PER_RUN_LIMIT;
+
+  useEffect(() => {
+    if (!isFeatureFlagsLoading && field.value > maxLimit) {
+      field.onChange(maxLimit);
+    }
+  }, [isFeatureFlagsLoading, maxLimit, field]);
+
+  const displayLabel = isFeatureFlagsLoading
+    ? "Max steps settings"
```

Comment:
I would avoid the useEffect and load that first then putting the value into the default value